classes/Category.php

```diff
@@ -1435,44 +1481,55 @@ public function getParentsCategories($idLang = null)
-			FROM `'._DB_PREFIX_.'category` c
+
+        $sqlAppend = 'FROM `'._DB_PREFIX_.'category` c
 			LEFT JOIN `'._DB_PREFIX_.'category_lang` cl
 				ON (c.`id_category` = cl.`id_category`
-				AND `id_lang` = '.(int) $idLang.Shop::addSqlRestrictionOnLang('cl').')';
-            if (Shop::isFeatureActive() && Shop::getContext() == Shop::CONTEXT_SHOP) {
-                $sql .= ' LEFT JOIN `'._DB_PREFIX_.'category_shop` cs ON (c.`id_category` = cs.`id_category` AND cs.`id_shop` = '.(int) $idShop.')';
-            }
-            $sql .= ' WHERE c.`id_category` = '.(int) $idCurrent;
-            if (Shop::isFeatureActive() && Shop::getContext() == Shop::CONTEXT_SHOP) {
-                $sql .= ' AND cs.`id_shop` = '.(int) $context->shop->id;
-            }
-            $rootCategory = Category::getRootCategory();
-            if (Shop::isFeatureActive() && Shop::getContext() == Shop::CONTEXT_SHOP
-                && (!Tools::isSubmit('id_category') || (int) Tools::getValue('id_category') == (int) $rootCategory->id || (int) $rootCategory->id == (int) $context->shop->id_category)) {
-                $sql .= ' AND c.`id_parent` != 0';
-            }
+                    AND `id_lang` = '.(int) $idLang.Shop::addSqlRestrictionOnLang('cl').')';
+        if (Shop::isFeatureActive() && Shop::getContext() === Shop::CONTEXT_SHOP) {
+            $sqlAppend .= ' LEFT JOIN `'._DB_PREFIX_.'category_shop` cs '.
+                'ON (c.`id_category` = cs.`id_category` AND cs.`id_shop` = '.(int) $idShop.')';
+        }
+        if (Shop::isFeatureActive() && Shop::getContext() === Shop::CONTEXT_SHOP) {
+            $sqlAppend .= ' AND cs.`id_shop` = '.(int) $context->shop->id;
+        }
+        $rootCategory = Category::getRootCategory();
+        if (Shop::isFeatureActive() && Shop::getContext() === Shop::CONTEXT_SHOP
+            && (!Tools::isSubmit('id_category')
+                || (int) Tools::getValue('id_category') == (int) $rootCategory->id
+                || (int) $rootCategory->id == (int) $context->shop->id_category)) {
+            $sqlAppend .= ' AND c.`id_parent` != 0';
+        }
 
-            $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow($sql);
+        $categories = [];
 
-            if ($result) {
-                $categories[] = $result;
-            } elseif (!$categories) {
-                $categories = array();
-            }
-            if (!$result || ($result['id_category'] == $context->shop->id_category)) {
-                return $categories;
-            }
-            $idCurrent = $result['id_parent'];
+        $treeInfo = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow(
+            'SELECT c.`nleft`, c.`nright`  '.$sqlAppend.' WHERE c.`id_category` = '.(int) $idCurrent
+        );
+        $rootTreeInfo = Db::getInstance(_PS_USE_SQL_SLAVE_)->getRow(
+            'SELECT c.`nleft`, c.`nright` FROM `'._DB_PREFIX_.'category` c 
+            WHERE c.`id_category` = '.(int) $context->shop->id_category
+        );
```

Comment:
It seems this request is not mandatory `if (!empty($treeInfo))`, and can be moved in the `if` block?